pkgs/applications/networking/msmtp/paths.patch

```diff
@@ -1,98 +1,64 @@
-diff --git a/scripts/msmtpq/msmtp-queue b/scripts/msmtpq/msmtp-queue
-index 1dc220d..d834241 100755
---- a/scripts/msmtpq/msmtp-queue
-+++ b/scripts/msmtpq/msmtp-queue
-@@ -27,4 +27,4 @@
- ## change the below line to be
- ##   exec /path/to/msmtpq --q-mgmt
- 
--exec msmtpq --q-mgmt "$1"
-+exec @msmtpq@ --q-mgmt "$1"
 diff --git a/scripts/msmtpq/msmtpq b/scripts/msmtpq/msmtpq
-index bdb4fb8..1363a67 100755
+index 1b39fc6..4baa19b 100755
 --- a/scripts/msmtpq/msmtpq
 +++ b/scripts/msmtpq/msmtpq
-@@ -59,7 +59,7 @@ err() { dsp '' "$@" '' ; exit 1 ; }
- ##   enter the location of the msmtp executable  (no quotes !!)
- ##   e.g. ( MSMTP=/path/to/msmtp )
- ##   and uncomment the test for its existence
--MSMTP=msmtp
-+MSMTP=@msmtp@
- #[ -x "$MSMTP" ] || \
- #  log -e 1 "msmtpq : can't find the msmtp executable [ $MSMTP ]"   # if not found - complain ; quit
- ##
-@@ -70,9 +70,9 @@ MSMTP=msmtp
+@@ -70,8 +70,8 @@ MSMTP=msmtp
  ##            ( chmod 0700 msmtp.queue )
  ##
  ## the queue dir - modify this to reflect where you'd like it to be  (no quotes !!)
 -Q=~/.msmtp.queue
 -[ -d "$Q" ] || mkdir -m 0700 "$Q" || \
--  err '' "msmtpq : can't find or create msmtp queue directory [ $Q ]" ''     # if not present - complain ; quit
 +Q=${MSMTP_QUEUE:-~/.msmtp.queue}
 +[ -d "$Q" ] || mkdir -m 0700 -p "$Q" || \
-+   err '' "msmtpq : can't find or create msmtp queue directory [ $Q ]" ''     # if not present - complain ; quit
+   err '' "msmtpq : can't find or create msmtp queue directory [ $Q ]" ''     # if not present - complain ; quit
  ##
  ## set the queue log file var to the location of the msmtp queue log file
- ##   where it is or where you'd like it to be
-@@ -84,7 +83,10 @@ Q=~/.msmtp.queue
+@@ -84,7 +84,10 @@ Q=~/.msmtp.queue
  ##     (doing so would be inadvisable under most conditions, however)
  ##
  ## the queue log file - modify (or comment out) to taste  (but no quotes !!)
 -LOG=~/log/msmtp.queue.log
 +LOG=${MSMTP_LOG:-~/log/msmtp.queue.log}
-+test -d "$(dirname $LOG)" || mkdir -p "$(dirname $LOG)"
++[ -d "$(dirname "$LOG")" ] || mkdir -p "$(dirname "$LOG")"
 +
 +JOURNAL=@journal@
  ## ======================================================================================
  
  ## msmtpq can use the following environment variables :
-@@ -108,7 +110,7 @@ LOG=~/log/msmtp.queue.log
+@@ -108,7 +111,7 @@ LOG=~/log/msmtp.queue.log
  ##
  #EMAIL_CONN_NOTEST=y                 # deprecated ; use below var
  #EMAIL_CONN_TEST={x| |p|P|n|s}       # see settings above for EMAIL_CONN_TEST
 -EMAIL_CONN_TEST=n
-+EMAIL_CONN_TEST=@test@
++EMAIL_CONN_TEST=${MSMTP_CONN_TEST:-n}
```

Comment:
I would just not set `EMAIL_CONN_TEST` at all. According to the `connect_test()` function, not setting `EMAIL_CONN_TEST` is supposed to be the same as setting `EMAIL_CONN_TEST` to `p`. [The line that set it to `n` just recently got removed from the upstream repo.](https://git.marlam.de/gitweb/?p=msmtp.git;a=commitdiff;h=f144a9be87c4dfa7d3750a76e3f288de0d8a8253)

Reply:
With the recent upstream changes, that makes sense - also means patching less which is definitely a plus!